extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java

```diff
@@ -0,0 +1,696 @@
+        httpSourceClientAndServer
+                .when(
+                        givenGetRequest(),
+                        exactly(2)
+                )
+                .error(
+                        withDropConnection()
+                );
+
+        // Next call to HTTP Source returns a valid response.
+        var body = FAKER.internet().uuid();
+        httpSourceClientAndServer
+                .when(
+                        givenGetRequest(),
+                        once()
+                )
+                .respond(
+                        withResponse(HttpStatusCode.OK_200, body)
+                );
+
+        // HTTP Sink Request & Response
+        httpSinkClientAndServer
+                .when(
+                        givenPostRequest(body),
+                        once()
+                )
+                .respond(
+                        withResponse(HttpStatusCode.OK_200)
+                );
+
+        // Act & Assert
+        // Initiate transfer
+        initiateTransfer(processId);
+
+        // Wait for transfer to be completed.
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                validateTransferResult(processId)
+        );
+        // Verify HTTP Source server called exactly 3 times.
+        httpSourceClientAndServer.verify(
+                givenGetRequest(),
+                VerificationTimes.exactly(3)
+        );
+        // Verify HTTP Sink server called exactly once.
+        httpSinkClientAndServer.verify(
+                givenPostRequest(body),
+                VerificationTimes.once()
+        );
+    }
+
+    /**
+     * Validate DPF doesn't retry to deliver the data if the sink drops the connection.
+     */
+    @Test
+    public void transfer_sinkDropsConnection_noRetry() {
+        // Arrange
+        // HTTP Source Request & Response
+        var body = FAKER.internet().uuid();
+        var processId = FAKER.internet().uuid();
+        httpSourceClientAndServer
+                .when(
+                        givenGetRequest(),
+                        once()
+                )
+                .respond(
+                        withResponse(HttpStatusCode.OK_200, body)
+                );
+
+        // HTTP sink drops the connection.
+        httpSinkClientAndServer
+                .when(
+                        givenPostRequest(body),
+                        once()
+                )
+                .error(
+                        withDropConnection()
+                );
+
+        // Act & Assert
+        // Initiate transfer
+        initiateTransfer(processId);
+
+        // Wait for transfer to be completed.
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                validateTransferResult(processId)
+        );
+
+        // Verify HTTP Source server called exactly once.
+        httpSourceClientAndServer.verify(
+                givenGetRequest(),
+                VerificationTimes.once()
```

Comment:
static import?

Reply:
we have other one import static org.mockserver.matchers.Times.once; for request expectation, that's why calling with class name here.